divvy/client.py

```diff
@@ -28,12 +28,23 @@ def __init__(self, host='localhost', port=8321,
             retry_on_timeout=retry_on_timeout
         )
 
-    def hit(self, **kwargs):
-        """Perform a check-and-decrement of quota. Zero or more key-value pairs
-        specify the operation being performed, and will be evaluated by the
-        server against its configuration."""
-        if len(kwargs) == 0:
-            raise InputError("hit() requires at least one keyword argument")
+    def check_rate_limit(self, **kwargs):
+        """
+        Perform a check-and-decrement of quota.
+
+        Args:
+             **kwargs: Zero or more key-value pairs to specify the operation
+                being performed, which will be evaluated by the server against
+                its configuration.
+
+        Returns:
+            divvy.Response, a namedtuple with these fields:
+                is_allowed: one of true, false, indicating whether quota was
+                    available.
+                current_credit: number of credit(s) available at the end of
+                    this command.
+                next_reset_seconds: time, in seconds, until credit next resets.
+        """
         cmd = self.translator.build_hit(**kwargs)
         self.connection.send(cmd)
 
```

Comment:
(line below) -- is it possible for recv() to only get partial results? does the memcached or redis client handle that case?

Reply:
The entire response can't be more than about 40 bytes: it's always `OK <bool> <int> <int> <int>`. Even with the largest ints, you're looking at approximately 41 bytes. The client is using a 1024-byte buffer so there's plenty of room in a single `recv()`.

The Redis client makes a single `socket.recv(1024)` so I'm pretty sure this is the right way to do it.

Reply:
Ok if redis does it that way I feel better about it. I was wondering if it was possible that a recv would only get (say) 20 bytes, and recv() would need to be called again one or more times to get the rest.